### ISSUE TITLE
Workflow with FAILED status 

### ISSUE THREAD

**Issue by alexei-led** [sdminonne/workflow-controller#37]
Currently, Workflow can have one of two possible statuses: 
1. `WorkflowComplete` - when all its steps are completed
2. `WorkflowFailed` - when "Deadline" is exceeded

IMHO, it should be **failed** also when one of workflow steps fails (Job fails).

**Comment by sdminonne** [comment]
Yeah, in general, I concur. Problem is what are we going to do when a Job failed?

1. Should we implement a retry policy? I think we should.
2. When a job failed should we stop all the workflow? I think we should
3. Remove the all the workflows? I think we should not.

Thoughts?
@clamoriniere1A: ideas?

**Comment by alexei-led** [comment]
@sdminonne when Job fail, after `backoffLimit` retries, we can stop the Workflow (if a user wants to, by specifying some tag). 
And I think we should not remove the failed workflow, user can do it later if wants to. Keeping workflow generated jobs/pods will allow inspecting "Failure" and maybe "fixing" workflow for next run.

**Comment by sdminonne** [comment]
@alexei-led  agree. Need to put together a proposal for the tags.
